kive/librarian/ajax.py

```diff
@@ -170,5 +170,9 @@ def download(self, request, pk=None):
         Handles downloading of the Dataset.
         """
         dataset = self.get_object()
+        dataset_handle = dataset.get_open_file_handle()
 
-        return build_download_response(dataset.dataset_file)
+        if dataset_handle is not None:
+            return build_download_response(dataset_handle)
+        else:
+            return Response(status=500, msg="Couldn't find dataset file")
```

Comment:
In order to get a JSON response with a 500 status code, I think you need to raise an `APIException`. Check to be sure.

Reply:
Consider adding a unit test for this error handling.